scripts/zkai_may2026_competitor_metric_matrix_gate.py

```
@@ -0,0 +1,629 @@
+    return _parse_json_bytes(path, read_source_bytes(path, "JSON"))
+
+
+def _reject_json_constant(value: str) -> None:
+    raise ValueError(f"non-finite JSON constant: {value}")
+
+
+def _parse_tsv_bytes(path: pathlib.Path, raw: bytes) -> list[dict[str, str]]:
+    try:
+        text = raw.decode("utf-8")
+    except UnicodeDecodeError as err:
+        raise CompetitorMetricMatrixError(f"failed to load TSV source {path}: {err}") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)